src/ngraph_rewrite_for_tracking.cc

```diff
@@ -83,6 +85,14 @@ Status RewriteForTracking(Graph* graph) {
 
         replacement->set_assigned_device_name(node->assigned_device_name());
 
+        // Add edge from the input nodes (to the variable node (NGraphVariable))
+        // to the new replacement node (also of type NGraphVariable)
+        for (auto edge : node->in_edges()) {
+          NGRAPH_VLOG(4) << "Replacing: " << edge->DebugString();
+          graph->AddEdge(edge->src(), edge->src_output(), replacement,
+                         edge->dst_input());
+        }
+
```

Comment:
Should it be AddEdge or UpdateEdge? (depending on whether we want to retain the original edge)
in case of addedge we are not really replacing the edge, so the print line could be:
`NGRAPH_VLOG(4) << "Cloning: " << edge->DebugString();` instead of `Replacing`

Reply:
NVM, above comment is invalid